subprojects/dependency-management/src/main/java/org/gradle/api/internal/filestore/url/UrlFileStore.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.filestore.url;
+
+import org.gradle.api.Transformer;
+import org.gradle.api.internal.file.TemporaryFileProvider;
+import org.gradle.internal.resource.local.GroupedAndNamedUniqueFileStore;
+import org.gradle.internal.resource.local.PathKeyFileStore;
+
+import java.net.URL;
+
+public class UrlFileStore extends GroupedAndNamedUniqueFileStore<URL> {
```

Comment:
This isn't quite right as we aren't considering the port, query params or fragment. We should just hash the canonical URL string.